src/main/host/syscall/handler/timerfd.rs

```diff
@@ -0,0 +1,244 @@
+                it_value: remaining.try_into().unwrap(),
+                it_interval: interval.try_into().unwrap(),
+            };
+
+            // Write the result.
+            runnable.memory_borrow_mut().write(value_ptr, &result)?;
+        }
+
+        Ok(())
+    }
+
+    #[log_syscall(/* rv */ libc::c_int, /* fd */ libc::c_int, /* flags */ libc::c_int,
+                  /* new_value */ *const libc::c_void, /* old_value */ *const libc::c_void)]
+    pub fn timerfd_settime(
+        ctx: &mut SyscallContext,
+        fd: libc::c_int,
+        flags: libc::c_int,
+        new_value_ptr: ForeignPtr<libc::itimerspec>,
+        old_value_ptr: ForeignPtr<libc::itimerspec>,
+    ) -> SyscallResult {
+        let Some(flags) = TimerSetTimeFlags::from_bits(flags) else {
+            log::warn!("Invalid timerfd_settime flags: {}", flags);
+            return Err(Errno::EINVAL.into());
+        };
+
+        // Get the TimerFd object.
+        let file = get_cloned_file(ctx, fd)?;
+        let File::TimerFd(ref timerfd) = file else {
+            return Err(Errno::EINVAL.into());
+        };
+
+        // Read in the new value from the plugin.
+        let new_value = {
+            // Need to be able to read the new timer config.
+            if new_value_ptr.is_null() {
+                return Err(Errno::EFAULT.into());
+            }
+
+            // Error if the process disappeared.
+            let Some(runnable) = ctx.objs.process.borrow_runnable() else {
+                return Err(Errno::EINVAL.into());
+            };
+
+            // this is a clippy false-positive
+            #[allow(clippy::let_and_return)]
+            let val = runnable.memory_borrow().read(new_value_ptr)?;
+            val
+        };
+
+        // Verify a valid range for new_time nanosecond vals.
+        if new_value.it_value.tv_nsec < 0
+            || new_value.it_value.tv_nsec >= 999_999_999
+            || new_value.it_interval.tv_nsec < 0
+            || new_value.it_interval.tv_nsec >= 999_999_999
+        {
+            return Err(Errno::EINVAL.into());
+        }
+
+        // First, write out the old time if requested.
+        if !old_value_ptr.is_null() {
+            // Old value is always relative, even if TFD_TIMER_ABSTIME is set.
+            Self::timerfd_gettime_helper(ctx, timerfd, old_value_ptr)?;
+        }
+
+        // Now we can adjust the timer with the new_value.
+        if new_value.it_value.tv_sec == 0 && new_value.it_value.tv_nsec == 0 {
+            // A value of 0 disarms the timer; it_interval is ignored.
+            TimerFd::disarm_timer(timerfd);
+            log::trace!("TimerFd {} disarmed", fd);
+        } else {
+            // Need to arm the timer, value may be absolute or relative.
+            let value: SimulationTime = new_value.it_value.try_into().unwrap();
+            let interval: SimulationTime = new_value.it_interval.try_into().unwrap();
+            let now = Worker::current_time().unwrap();
```

Comment:
To reviewer: please check if these unwraps are appropriate here.

Reply:
See comment above. These could panic on negative `tv_sec` values.